modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java

```diff
@@ -568,12 +585,80 @@ public void close() {
 
     @Override
     public CompletableFuture<Void> snapshot(Path snapshotPath) {
-        throw new UnsupportedOperationException();
+        synchronized (mux) {
+            try {
+                Files.createDirectories(snapshotPath);
+
+                Path snapshotFile = snapshotPath.resolve(SimpleInMemoryKeyValueStorageSnapshot.FILE_NAME);
+
+                assertTrue(IgniteUtils.deleteIfExists(snapshotFile), snapshotFile.toString());
+
+                Files.createFile(snapshotFile);
+
+                var revsIdxCopy = new HashMap<Long, Map<byte[], ValueSnapshot>>();
```

Comment:
May I suggest a little more idiomatic approach?

```
Map<Long, Map<byte[], ValueSnapshot>> revsIdxCopy = revsIdx.entrySet().stream()
        .collect(toMap(
                Map.Entry::getKey,
                revIdxEntry -> revIdxEntry.getValue()
                        .entrySet()
                        .stream()
                        .collect(toMap(Map.Entry::getKey, e -> new ValueSnapshot(e.getValue())))
        ));
```

Reply:
why not.